github/users_gpg_keys_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2016 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestUsersService_ListGPGKeys(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/user/gpg_keys", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testHeader(t, r, "Accept", mediaTypeGitSigningPreview)
+		fmt.Fprint(w, `[{"id":1,"primary_key_id":2}]`)
+	})
+
+	keys, _, err := client.Users.ListGPGKeys()
+	if err != nil {
+		t.Errorf("Users.ListGPGKeys returned error: %v", err)
+	}
+
+	want := []GPGKey{{ID: Int(1), PrimaryKeyID: Int(2)}}
+	if !reflect.DeepEqual(keys, want) {
+		t.Errorf("Users.ListGPGKeys = %+v, want %+v", keys, want)
+	}
+}
+
+func TestUsersService_GetGPGKey(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/user/gpg_keys/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testHeader(t, r, "Accept", mediaTypeGitSigningPreview)
+		fmt.Fprint(w, `{"id":1}`)
+	})
+
+	key, _, err := client.Users.GetGPGKey(1)
+	if err != nil {
+		t.Errorf("Users.GetGPGKey returned error: %v", err)
+	}
+
+	want := &GPGKey{ID: Int(1)}
+	if !reflect.DeepEqual(key, want) {
+		t.Errorf("Users.GetGPGKey = %+v, want %+v", key, want)
+	}
+}
+
+func TestUsersService_CreateGPGKey(t *testing.T) {
+	setup()
+	defer teardown()
+
+	input := `
+-----BEGIN PGP PUBLIC KEY BLOCK-----
+Comment: GPGTools - https://gpgtools.org
+
+mQINBFcEd9kBEACo54TDbGhKlXKWMvJgecEUKPPcv7XdnpKdGb3LRw5MvFwT0V0f
+...
+=tqfb
+-----END PGP PUBLIC KEY BLOCK-----`
+
+	mux.HandleFunc("/user/gpg_keys", func(w http.ResponseWriter, r *http.Request) {
+		gpgKey := &struct {
+			ArmoredPublicKey *string `json:"armored_public_key,omitempty"`
+		}{}
+		json.NewDecoder(r.Body).Decode(gpgKey)
```

Comment:
Minor style nitpick. Since you're just creating a blank `gpgKey`, I would write the above as:

```Go
var gpgKey struct {
	ArmoredPublicKey *string `json:"armored_public_key,omitempty"`
}
json.NewDecoder(r.Body).Decode(&gpgKey)
```

I think that's a little easier to read and see that `gpgKey` is just a zero value. It's up to you if you want to take this suggestion.